block-producer/src/store/mod.rs

```diff
@@ -88,6 +88,8 @@ impl DefaultStore {
 
 #[async_trait]
 impl ApplyBlock for DefaultStore {
+    #[allow(clippy::blocks_in_conditions)] // Workaround of `instrument` issue
+    #[instrument(target = "miden-block-producer", skip_all, err(level = "warn"))]
```

Comment:
@bobbinth default severity for errors is `error`, I decided to lower severity here, because of retrying in a case of error. But I can revert it, if you think, it would be better. I think, we shouldn't lower severity for all errors, only in particular cases.

Reply:
Ah yes - let's keep errors at `error` level.

For this error - what do you mean by "retrying in a case of error" - does this work differently from other functions?

Reply:
> For this error - what do you mean by "retrying in a case of error" - does this work differently from other functions?

You're right, we retry batch/block building, so I'll return it to `error`. Thank you!